hack/run-e2e-kind.sh

```diff
@@ -27,49 +27,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export ROOT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
+export ROOT_DIR="$(dirname "$(dirname "$(readlink -fm "$0")")")"
```

Comment:
This change is mostly, cosmetic as it eliminates the `..` from the directory name. The double `dirname` actives that. `readlink` resolves to the actual directory ensuring that any symbolic links are followed and returns the name of the directory. Here's [a nice article](https://linuxhint.com/readlink-command-linux/) explaining the usage.

Reply:
Very nice, thank you.  Followup question regarding the diffs.  See below:
On my mac:
```
$ pwd
/Users/darroyo/git_workspaces/github.com/dmatch01/IBM/multi-cluster-app-dispatcher
$
$ RD=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
$
$ echo $RD
/Users/darroyo/git_workspaces/github.com/dmatch01/IBM/multi-cluster-app-dispatcher/..
$
RD="$(dirname "$(dirname "$(readlink -f "$0")")")"
$
$ echo $RD
.
$
```
Is this result diff. intended:
`Users/darroyo/git_workspaces/github.com/dmatch01/IBM/multi-cluster-app-dispatcher/..`
vs
`.`?

Reply:
No, I think there's a missing argument to `readlink` in your example, here should be an `n`. Here's a simple script highlighting the differences:

```bash
#!/bin/bash
ROOT_DIR_1=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
ROOT_DIR_2="$(dirname "$(dirname "$(readlink -fn "$0")")")"

echo "ROOT_DIR_1: ${ROOT_DIR_1}"
echo "ROOT_DIR_2: ${ROOT_DIR_2}"
```

If I invoke from this local directory `/Users/laurentiu.bradin/work/bin` it shows this output:
```
ROOT_DIR_1: /Users/laurentiu.bradin/work/bin/..
ROOT_DIR_2: /Users/laurentiu.bradin/work
```

There's a bug I have introduced (cut paste error) in the e2e script. Thanks for the question.  There's a fly in the ointment when using the readlink on macOS. It's not installed by default and it will require installing additional packages via `brew`. 

Do you want to remove the usage of the readlink command or document the additional package installation?

Reply:
Thanks for catching that.  I'm still getting the same results:
```
$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin21)
Copyright (C) 2007 Free Software Foundation, Inc.
$
$ ROOT_DIR_1=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
$
$ ROOT_DIR_2="$(dirname "$(dirname "$(readlink -fn "$0")")")"
$ echo $ROOT_DIR_1 
/Users/darroyo/git_workspaces/github.com/dmatch01/IBM/multi-cluster-app-dispatcher/..
$
$ echo $ROOT_DIR_2
.
$ pwd
/Users/darroyo/git_workspaces/github.com/dmatch01/IBM/multi-cluster-app-dispatcher
$
```

> Do you want to remove the usage of the readlink command or document the additional package installation?

I'm good for doc. updates.

Reply:
Here ya go:
```
bash-3.2$ which readlink
/usr/bin/readlink
bash-3.2$
bash-3.2$ echo $0
bash
bash-3.2$
bash-3.2$ readlink -fn $0
bash-3.2$
bash-3.2$ readlink -f $0
bash-3.2$
bash-3.2$ readlink -fn $0
bash-3.2$ 

```